#171784089 Edit User Profile Information

Francois-MUGOROZI opened the pull request:
#### What does this PR do?
Enable user to update profile info
#### Description of Task to be completed?
- Create update user profile endpoint
- Create update user photo endpoint 
- Create get user profile endpoint
#### How should this be manually tested?
Give that the user has signed up (refer to signup feature); clone the repo
cd into it and run npm install then npm run dev.
using postman, send a patch request at localhost:5000/api/v1/profile with data in the body.
and to get user profile, send get request at localhost:5000/api/v1/profile with an authorization token in headers
Refer to the API docs by browsing: http://localhost:5000/api-docs/
Make sure you have Postgres database and create .env refer to .env.example

#### What are the relevant pivotal tracker stories?
[171784089](https://www.pivotaltracker.com/n/projects/2438874/stories/171784089)

irfiacre commented on server/middlewares/authorization.js:
```diff
@@ -0,0 +1,22 @@
+const util = new Util();
+
+// eslint-disable-next-line consistent-return
+export const authorized = async (req, res, next) => {
```
Good work on the authorization, but
I would like to suggest maybe use a name that is giving more meaning like `authorizationCheck, checkAuthorization`, or others.

Francois-MUGOROZI replied:
That's what I was suggesting if we can have a common middleware for authorization because the application has to use the same authorization way.

irfiacre commented on server/controllers/usercontroller.js:
```diff
@@ -118,6 +123,56 @@ class User {
+
+  static async updateUserProfile(req, res) {
+    const userId = req.user.id;
+    if (req.body.photoUrl) {
```
It could be better if all checks were in the middleware

irfiacre reviewed (commented):
Nice work

bdushimi commented on logs/all.log.15-04-2020:
```diff
@@ -0,0 +1,33 @@
+{"service":"user-service","level":"info","message":"METHOD: POST, and the URL: /api/v1/trips, STATUSCODE: 200 IP-address: ::ffff:127.0.0.1"}
```
@Francois-MUGOROZI kindly add your local log files to gitignore.

bdushimi commented on package.json:
```diff
@@ -24,7 +24,10 @@
+    "devtable": " npm run drop && npm run createTables && npx sequelize-cli db:seed:all",
+    "test1": "set NODE_ENV=test && npm run drop:test && npm run testTables && npx sequelize-cli db:seed:all --env test && nyc --reporter=html --reporter=text mocha --require @babel/register --timeout 10000 ./server/tests/*.test.js --exit",
+    "testTables": "npx sequelize-cli db:migrate --env test",
+    "drop:test": "npx sequelize-cli db:migrate:undo:all --to 20200331142748-create-users.js --env test"
```
@Francois-MUGOROZI Kindly check and use the scripts currently on develop branch. Some of the above scripts are redundant and not scalable.

bdushimi commented on server/controllers/usercontroller.js:
```diff
@@ -102,12 +105,14 @@ class User {
       expiresIn: '4h',
     });
 
+    userInfo.id = undefined;
```
@Francois-MUGOROZI any reason why the userInfo.id gets assigned undefined yet the user exists in the database?

bdushimi commented on server/controllers/usercontroller.js:
```diff
@@ -118,6 +123,56 @@ class User {
+    }
+    await userService.updateAtt(req.body, { id: userId });
+    const message = 'Profile updated successfuly';
+    infoLogger(req, 200, message);
```
@Francois-MUGOROZI kindly log errors only; responses with 4xx or 5xx status code. Apply this elsewhere applicable.

bdushimi commented on server/controllers/usercontroller.js:
```diff
@@ -118,6 +123,56 @@ class User {
+  static async updateUserProfile(req, res) {
+    const userId = req.userData.id;
+    if (req.body.photoUrl) {
+      const photoExist = await photoService.findByProp({ ownerId: userId });
```
@Francois-MUGOROZI I think we need to discuss how you implemented this service, it is not how we once discussed about it on the call.

Francois-MUGOROZI committed:
ft(update user profile): add user profile update factionality
- added profile update endpoint
-added tests ad documentation
-get latest changes and mplement feefback
[Finish #171784089]
